modules/azurerm_vm/main.tf

```diff
@@ -7,10 +7,11 @@ resource "azurerm_public_ip" "vm_ip" {
   name                = "${var.name}-public_ip"
   location            = var.azure_rg_location
   resource_group_name = var.azure_rg_name
-  allocation_method   = "Static"
+  allocation_method   = var.public_ip_allocation_method
   sku                 = var.vm_zone == null ? "Basic" : "Standard"
   zones               = var.vm_zone == null ? [] : [var.vm_zone]
   tags                = var.tags
+  domain_name_label   = var.public_ip_domain_name_label
```

Comment:
Is the `domain_name_label` always set? Is it optional? Maybe this should be a local. Is this domain only relevant to the VM's being created? Maybe the variable is focused on that. I would drop the `label` element of the variable. It's a domain name for the machine.

Reply:
I left the label portion of the name since that is what terraform has (I have tried to keep to the terraform conventions as much as possible). Right now this works by either passing in nothing (if it is not set) or an actual value. It could be rewritten to something more like the following for better readability.

domain_name_label   = var.public_ip_domain_name != "" ? var.public_ip_domain_name : ""